radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/handlers/OlympiaStateHandler.java

```diff
@@ -0,0 +1,135 @@
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.api.core.handlers;
+
+import com.google.inject.Inject;
+import com.radixdlt.constraintmachine.SubstateDeserialization;
+import com.radixdlt.environment.Runners;
+import com.radixdlt.modules.ModuleRunner;
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.statecomputer.forks.Forks;
+import com.radixdlt.stateir.OlympiaStateIRSerializer;
+import com.radixdlt.stateir.StateIRConstructor;
+import com.radixdlt.store.EngineStore;
+import com.radixdlt.utils.Bytes;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+import org.xerial.snappy.Snappy;
+
+public final class OlympiaStateHandler implements HttpHandler {
+  private final EngineStore<LedgerAndBFTProof> engineStore;
+  private final SubstateDeserialization substateDeserialization;
+
+  // TODO: A hack for testing, remove me
+  private final Map<String, ModuleRunner> moduleRunners;
```

Comment:
We have things like that in the codebase already and I really don't think we should be adding more. Usually later equals never https://yiming.dev/clipping/2019/03/21/le-blanc's-law-a-k-a-later-equals-never/#:~:text=LeBlanc's%20Law%3A%20%22Later%20equals%20Never,fast%20you%20can%20deliver%20value.

Reply:
As mentioned in my comment, this endpoint is really only meant for testing. I guess we don't really need it on main, I can remove it from the PR and keep it on a separate branch, if you prefer.

Reply:
I think I wouldn't merge it into main now. We can add it later when the process will better determined. Maybe we can test it with an integration test for now?

Reply:
(removed)